### ISSUE TITLE
Existing device with fresh config.json will not properly re-provision

### ISSUE THREAD

**Issue by toochevere** [balena-os/balena-supervisor#1737]
Bug:  If you take an existing device and copy over a fresh `config.json`, it will attempt to reprovision but will not do so correctly.  The device ends up offline with a partially provisioned `config.json` unless the supervisor database is also deleted.

Steps to reproduce:
1. On an existing device, copy over a new config.json
2. Restart the device

After letting the device have a minute to try to provision, look at the `config.json`.  It will appear to have provisioned, but `apiKey` is still present.  Additionally, it remains offline in the dashboard.  

If the supervisor is stopped and it's database deleted at the time that the new config.json is copied over, then the provisioning succeeds.

**Comment by 20k-ultra** [comment]
I understand this is a bug that we'll fix but, do you know if this is a common thing people do ? delete config.json to "re-provision" ?

**Comment by cywang117** [comment]
I don't think this is a bug though -- we've warned users that doing things with config.json can be dangerous in the docs, and if they're doing this despite that, I don't know what else we can do to warn them. If there's enough of a need to "re-provision" a device we should expose it as an official action for the user. If we go down the path of respecting user-made changes to the config.json, we can reach a situation where:

1) With bad config.json changes, issues are hard to debug, because due to bad config.json, the device is offline/heartbeat only/etc.
2) We now have multiple sources of truth for configs - the cloud & config.json. The cloud should be telling the Supervisor what the right configs to set are. Of course, this point doesn't apply for unmanaged mode (I think there's a config-related bug in unmanaged mode right now: #1727)

**Comment by pipex** [comment]
@20k-ultra @cywang117 I asked PJ to create this issue. Is not really a bug, but this means the supervisor is using multiple sources of truth to know when to provision (database and config.json), which may have unintended side effects. There might be an improvement to be made which I think will ultimately allow to simplify the supervisor code.
